### PR TITLE
move constants of internal autoscaling to its subclasses to resolve interdependence

### PR DESCRIPTION
**Pull request by mars1024**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

This pr moves some constants of internal autoscaling to its subclasses (v1, v2beat1, v2beat2), this can help to resolve interdependence of autoscaling classes above, and this can make it easy for others to import these constants.

This is the preheating commit to fix #74956 .

### CODE REVIEW

**Comment by mars1024** [comment]
@yue9944882

File: staging/src/k8s.io/api/autoscaling/v2beta2/types.go [hunk 1]
```diff
@@ -24,6 +24,12 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// DefaultCPUUtilization is the default value for CPU utilization, provided no other
+	// metrics are present.  This is here because it's used by v2beta2 defaulting logic.
+	DefaultCPUUtilization = 80
```
Comment (yue9944882) [inline staging/src/k8s.io/api/autoscaling/v2beta2/types.go]: default const value should be kept in internal autoscaling api

File: pkg/apis/autoscaling/annotations.go [hunk 2]
```diff
@@ -1,34 +0,0 @@
...
-const MetricStatusesAnnotation = "autoscaling.alpha.kubernetes.io/current-metrics"
-
-// HorizontalPodAutoscalerConditionsAnnotation is the annotation which holds the conditions
-// of an HPA when converting the `Conditions` field from autoscaling/v2beta1
-const HorizontalPodAutoscalerConditionsAnnotation = "autoscaling.alpha.kubernetes.io/conditions"
-
-// DefaultCPUUtilization is the default value for CPU utilization, provided no other
-// metrics are present.  This is here because it's used by both the v2beta1 defaulting
-// logic, and the pseudo-defaulting done in v1 conversion.
-const DefaultCPUUtilization = 80
```
Comment (yue9944882) [inline pkg/apis/autoscaling/annotations.go]: > default const value should be kept in internal autoscaling api

wait.. we should probably move this to `types.go`, then this file could be deleted

File: staging/src/k8s.io/api/autoscaling/v1/types.go [hunk 3]
```diff
@@ -22,6 +22,24 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// MetricSpecsAnnotation is the annotation which holds non-CPU-utilization HPA metric
+	// specs when converting the `Metrics` field from autoscaling/v2beta1 or autoscaling/v2beta2
+	MetricSpecsAnnotation = "autoscaling.alpha.kubernetes.io/metrics"
```
Comment (liggitt) [inline staging/src/k8s.io/api/autoscaling/v1/types.go]: why are these being moved externally? I would not expect anyone to be using these other than conversion

Comment (mars1024) [reply staging/src/k8s.io/api/autoscaling/v1/types.go]: OK, i've got your point, thanks.

**Pull request closed without merging**